### PR TITLE
Fixing change deck visibility endpoint

### PR DESCRIPTION
**Pull request by VanDavv**

### CODE REVIEW

File: backend/app/src/main/java/com/bls/patronage/api/DeckRepresentation.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
     @Length(min = 1, max = 50)
     final private String name;
     @NotNull
-    final private Boolean publicVisible;
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/api/DeckRepresentation.java]: Leave final. Use @JsonCreator on constructor to create this object.

Comment (VanDavv) [inline backend/app/src/main/java/com/bls/patronage/api/DeckRepresentation.java]: I fought that final is something unchangeabble, and we need to change this field, so i added setter and removed final. Do I have to use constructor instead?

File: backend/app/src/main/java/com/bls/patronage/resources/DeckResource.java [hunk 2]
```diff
@@ -66,9 +68,9 @@ public void changeStatus(@Auth User user,
                              @Valid @PathParam("access") BooleanParam access) {
 
         DeckRepresentation deck = new DeckRepresentation(
-                decksDAO.getDeckById(deckId.get(), user.getId()).getName(),
-                access.get()
+                decksDAO.getDeckById(deckId.get(), user.getId())
         );
+        deck.setPublicVisible(access.get());
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DeckResource.java]: Oh. Why do we blindly update this data in DB. We should do it only when entity has to change... Try to not mutate state of this object. I'll create new object with changed flag instead.

File: backend/app/src/main/java/com/bls/patronage/api/DeckRepresentation.java (hunk 1 above)
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/api/DeckRepresentation.java]: See other comment.

File: backend/app/src/main/java/com/bls/patronage/resources/DeckResource.java (hunk 2 above)
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DeckResource.java]: What should change: representation or DB model? `changeSattus` is void.

File: backend/app/src/main/java/com/bls/patronage/resources/DeckResource.java [hunk 3]
```diff
@@ -66,9 +68,9 @@ public void changeStatus(@Auth User user,
                              @Valid @PathParam("access") BooleanParam access) {
 
         DeckRepresentation deck = new DeckRepresentation(
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DeckResource.java]: Do we need DeckRepresentation here? I don't think so.

File: backend/app/src/main/java/com/bls/patronage/resources/DeckResource.java [hunk 4]
```diff
@@ -6,6 +6,8 @@
 import io.dropwizard.auth.Auth;
 import io.dropwizard.jersey.params.BooleanParam;
 import io.dropwizard.jersey.params.UUIDParam;
+import org.slf4j.Logger;
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DeckResource.java]: Unused?

Comment (VanDavv) [inline backend/app/src/main/java/com/bls/patronage/resources/DeckResource.java]: yes, just forgot to check "optimize imports" option

**Comment by mlotysz** [comment]
Thanks!

**Pull request merged**